### DISCUSSION TITLE
Depoly Tgindex Pro in Cpanel

### DISCUSSION THREAD

**Discussion by nacereddinedjebar** [rayanfer32/TgindexPro#26] [Q&A]
Dear Devs,

I'm trying to deploy the bot on cpanel hosting provider but i cannot determine the :
Application startup file.
Application Entry point.

Check attached screenshot, please need help and advice

Regards
![cpanel](https://user-images.githubusercontent.com/42089249/134297747-d5ab2ade-7108-4157-bb54-b867e97a7b8d.png)

**Comment by rayanfer32** [comment]
You need a server capable of running python applications, From my knowledge cpanel is mostly used for php or html deployment.

**Reply by nacereddinedjebar** [reply]
Thank you very much for your reply, i ttied to deploy a test python app on the server and it worked perfectly +  i also confirmed with the hosting provider that their servers are 100 % support python app.

**Reply by nacereddinedjebar** [reply]
does the app need django or flask ?

**Reply by pseudokawaii** [reply]
> 
> 
> does the app need django or flask ?

It uses aiohttp along with jinja2 template language.

**Reply by nacereddinedjebar** [reply]
Thank you very much for your reply i'll try more

**Reply by nacereddinedjebar** [reply]
Dears, which Python version are required?

**Reply by rayanfer32** [reply]
Python3.7 or above recommended.

**Reply by nacereddinedjebar** [reply]
@rayanfer32 thank you very much, no need for Django or Flask?

**Reply by rayanfer32** [reply]
no need

**Reply by nacereddinedjebar** [reply]
great <3 
![tgin](https://user-images.githubusercontent.com/42089249/134352977-9d460123-4217-4c09-8bab-f75675517e6f.png)

**Reply by rayanfer32** [reply]
Nice , you modified the ui ?

**Reply by nacereddinedjebar** [reply]
yes, and i added a button to export the list of files as a csv/excel file, + add the file sender name, size ..etc

**Reply by nacereddinedjebar** [reply]
@rayanfer32  the main startup file is run.py?

**Reply by rayanfer32** [reply]
cool, is your code public ?

**Reply by rayanfer32** [reply]
run.py is for noobs who try to run code without settings envs

**Reply by nacereddinedjebar** [reply]
yes, and i will publish it later on github

**Reply by nacereddinedjebar** [reply]
> run.py is for noobs who try to run code without settings envs

ah okay, cause i'm still trying to deploy it on my cpanel shared hosting, but i cannot determine what is the startup file.

**Reply by pseudokawaii** [reply]
@DeenCorp this is the entrypoint ~

https://github.com/rayanfer32/TgindexPro/blob/main/app/__main__.py

**Reply by nacereddinedjebar** [reply]
@pseudokawaii thank you very very much you saved me a lot of time

**Reply by rayanfer32** [reply]
Good , Now you have it working in cpanel too ryt :+1:
